test/statmodels.jl

```diff
@@ -7,27 +7,50 @@ v2 = ["Good", "Great", "Bad"]
 v3 = [1, 56, 2]
 v4 = [-12.56, 0.1326, 2.68e-16]
 v5 = [0.12, 0.3467, 1.345e-16]
-@test sprint(show, CoefTable(Any[v1, v2, v3, v4, v5],
-    ["Estimate", "Comments", "df", "t", "p"],
-    ["x1", "x2", "x3"], 5, 4)) == """
+ct = CoefTable(Any[v1, v2, v3, v4, v5],
+               ["Estimate", "Comments", "df", "t", "p"],
+               ["x1", "x2", "x3"], 5, 4)
+@test sprint(show, ct) == """
 ───────────────────────────────────────────────
          Estimate  Comments  df       t       p
 ───────────────────────────────────────────────
 x1    1.45666         Good    1  -12.56  0.1200
 x2  -23.14            Great  56    0.13  0.3467
 x3    1.56734e-13     Bad     2    0.00  <1e-15
 ───────────────────────────────────────────────"""
+@test length(ct) === 3
+@test eltype(ct) ==
+    NamedTuple{(:Name, :Estimate, :Comments, :df, :t, :p),
+               Tuple{String,Float64,String,Int64,Float64,Float64}}
+@test collect(ct) == [
+    (Name = "x1", Estimate = 1.45666, Comments = "Good", df = 1, t = -12.56, p = 0.12)
+    (Name = "x2", Estimate = -23.14, Comments = "Great", df = 56, t = 0.1326, p = 0.3467)
+    (Name = "x3", Estimate = 1.56734e-13, Comments = "Bad", df = 2, t = 2.68e-16, p = 1.345e-16)
+]
 
 Random.seed!(10)
 m = rand(3,4)
-@test sprint(show, CoefTable(m, ["Estimate", "Stderror", "df", "p"], [], 4)) == """
+ct = CoefTable(m, ["Estimate", "Stderror", "df", "p"], [], 4)
+@test sprint(show, ct) == """
 ──────────────────────────────────────────
      Estimate   Stderror        df       p
 ──────────────────────────────────────────
 [1]  0.112582  0.0566454  0.381813  0.8198
 [2]  0.368314  0.120781   0.815104  0.6699
 [3]  0.344454  0.179574   0.242208  0.4531
 ──────────────────────────────────────────"""
+@test length(ct) === 3
+@test eltype(ct) ==
+NamedTuple{(:Name, :Estimate, :Stderror, :df, :p),
+           Tuple{String,Float64,Float64,Float64,Float64}}
+@test collect(ct) == [
+    (Name = "[1]", Estimate = 0.11258244478647295, Stderror = 0.05664544616214151,
+     df = 0.38181274408522614, p = 0.8197779704008801)
```

Comment:
I'm hesitant as to whether it's better to always create a "Name" column even when no names have been specified. That makes the result more predictable, but that also creates a column which is nearly useless...

Reply:
I think it is better to keep a fixed schema for this.

Reply:
Well there isn't really a fixed schema since each package determines the column names and types. Actually, "Names" would be the only column which is guaranteed to be present. :-)

Reply:
Ah - in this case I would leave this out (i.e. rely on the schema provided by the package) and allow packages to specify the schema in full. Then actually we should check if packages provide a proper schema.